sample/weighted.go

```diff
@@ -0,0 +1,104 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this code is governed by a BSD-style
+// license that can be found in the LICENSE file
+
+package sample
+
+import "math/rand"
+
+// Weighted is a collection of items that can be selected with weighted probabilities
```

Comment:
I think this is tricky enough that it's better for the comment to express intent rather than just being a descriptive, something along the lines of

"Weighted is a type for sampling without replacement from a collection of items with weighted probabilites"